web/api/v1/api.go

```diff
@@ -338,14 +334,15 @@ func (api *API) query(r *http.Request) apiFuncResult {
 	qry, err := api.QueryEngine.NewInstantQuery(api.Queryable, r.FormValue("query"), ts)
 	if err != nil {
 		err = errors.Wrapf(err, "invalid parameter 'query'")
-		return apiFuncResult{nil, &apiError{errorBadData, err}, nil, nil}
+		return apiFuncResult{nil, &apiError{errorBadData, err}, nil}
 	}
+	defer qry.Close()
```

Comment:
This is incorrect, as we need to keep everything allocated until the API result has been sent out over HTTP - which is after this function and its defers have completed.